src/Cake.Common/IO/DirectoryAliases.cs

```diff
@@ -319,6 +320,49 @@ public static void EnsureDirectoryExists(this ICakeContext context, DirectoryPat
             }
         }
 
+        /// <summary>
+        /// Deletes the specified directory and its contents if it exists.
+        /// </summary>
+        /// <example>
+        /// <code>
+        /// EnsureDirectoryDoNotExists("./be/gone");
+        /// </code>
+        /// </example>
+        /// <param name="context">The context.</param>
+        /// <param name="path">The directory path.</param>
+        [CakeMethodAlias]
+        [CakeAliasCategory("DoNotExists")]
+        public static void EnsureDirectoryDoNotExists(this ICakeContext context, DirectoryPath path)
```

Comment:
I believe we should name it `EnsureDirectoryDoesNotExist` to be grammatically correct

```suggestion
        public static void EnsureDirectoryDoesNotExist(this ICakeContext context, DirectoryPath path)
```

Reply:
Totally on board with `DoesNot`. 

`Exists` is used for consistency with `EnsureDirectoryExists`. What do you suggest?

Reply:
They both look grammatically correct to me - though English is not my main language, so correct if I'm wrong:

- `EnsureDirectoryExists` - Ensure it exists (plural exists)
- `EnsureDirectoryDoesNotExist` - Ensure it does not exist (singular exist)